android/src/main/java/com/hoxfon/react/RNTwilioVoice/TwilioVoiceModule.java

```diff
@@ -492,6 +494,14 @@ public void initWithAccessToken(final String accessToken, Promise promise) {
             return;
         }
 
+        if(!checkPermissionForMicrophone()){
+            requestPermissionForMicrophone();
+            if(!checkPermissionForMicrophone())
```

Comment:
Please explain to me how this logic work, because it is not intuitive why `checkPermissionForMicrophone()` is called twice

Reply:
sorry about that , did it with VS code (which I don't regularly use) and somehow it pushed an old code.
Check out the new code and let me know if you agree with the check.